internal/app/client/conformance_client.go

```diff
@@ -0,0 +1,191 @@
+		// So any internal error that doesn't involve actually calling an RPC. In that case, then,
+		// this should just return an errors.New like the above cases
+		return &v1alpha1.ClientCompatResponse{
+			TestName: req.TestName,
+			Result: &v1alpha1.ClientCompatResponse_Error{
+				Error: &v1alpha1.ClientErrorResult{
+					Message: "method name " + req.Method + " does not exist",
+				},
+			},
+		}, nil
+	}
+}
+
+func (w *conformanceClientWrapper) unary(
+	ctx context.Context,
+	req *v1alpha1.ClientCompatRequest,
+) (*v1alpha1.ClientCompatResponse, error) {
+	ccResp := &v1alpha1.ClientCompatResponse{
+		TestName: req.TestName,
+	}
+	msg := req.RequestMessages[0]
+	ur := &v1alpha1.UnaryRequest{}
+	if err := msg.UnmarshalTo(ur); err != nil {
+		return nil, err
+	}
+
+	request := connect.NewRequest(ur)
+
+	// Add the specified request headers to the request
+	app.AddHeaders(req.RequestHeaders, request.Header())
+
+	var protoErr *v1alpha1.Error
+	var headers []*v1alpha1.Header
+	var trailers []*v1alpha1.Header
+	payloads := make([]*v1alpha1.ConformancePayload, 0, 1)
+
+	// Invoke the Unary call
+	resp, err := w.client.Unary(ctx, request)
+	if err != nil {
+		// If an error was returned, first convert it to a Connect error
+		// so that we can get the headers from the Meta property. Then,
+		// convert _that_ to a proto Error so we can set it in the response.
+		connectErr := app.ConvertErrorToConnectError(err)
+		headers = app.ConvertToProtoHeader(connectErr.Meta())
+		protoErr = app.ConvertConnectToProtoError(connectErr)
+	} else {
+		// If the call was successful, get the returned payloads
+		// and the headers and trailers
+		payloads = append(payloads, resp.Msg.Payload)
+		headers = app.ConvertToProtoHeader(resp.Header())
+		trailers = app.ConvertToProtoHeader(resp.Trailer())
+	}
+
+	ccResp.Result = &v1alpha1.ClientCompatResponse_Response{
+		Response: &v1alpha1.ClientResponseResult{
+			ResponseHeaders:  headers,
+			ResponseTrailers: trailers,
+			Payloads:         payloads,
+			Error:            protoErr,
+			ErrorDetailsRaw:  nil, // TODO
+		},
+	}
+
+	return ccResp, nil
+}
+
+func (w *conformanceClientWrapper) serverStream(
+	ctx context.Context,
+	req *v1alpha1.ClientCompatRequest,
+) (*v1alpha1.ClientCompatResponse, error) {
+	ccResp := &v1alpha1.ClientCompatResponse{
+		TestName: req.TestName,
+	}
+	msg := req.RequestMessages[0]
+	ssr := &v1alpha1.ServerStreamRequest{}
+	if err := msg.UnmarshalTo(ssr); err != nil {
+		return nil, err
+	}
+
+	request := connect.NewRequest(ssr)
+
+	// Add the specified request headers to the request
+	app.AddHeaders(req.RequestHeaders, request.Header())
+
+	stream, err := w.client.ServerStream(ctx, request)
+	// TODO - should this error be added to the clientcompatresponse or returned here?
+	// IMO, the error returned from this function represents an internal error independent
+	// of anything invoking the service but it's unclear what this err is vs. a stream.Err()
```

Comment:
See comments above, which hopefully clarify this. I think this should do what it's doing and return the error. And in `Invoke`, it gets stored into a the `error` field of a `*ClientResponseResult`.